test/nodes/test_retriever.py

```diff
@@ -80,6 +80,26 @@ def test_retrieval(retriever_with_docs: BaseRetriever, document_store_with_docs:
         )
         assert len(result) == 0
 
+    # test FilterRetriever with empty query using the run() method
+    if isinstance(retriever_with_docs, FilterRetriever):
+        result = retriever_with_docs.run(root_node="Query", query="", filters={"name": ["filename2"]})
+        assert len(result) == 2
+        result = result[0]
+        assert "documents" in result
+        result = result["documents"]
+        assert len(result) == 1
+        assert type(result[0]) == Document
+        assert result[0].content == "My name is Paul and I live in New York"
+        assert result[0].meta["name"] == "filename2"
+
+
+def test_retrieval_empty_query(document_store_with_docs: BaseDocumentStore):
+    # test with empty query using the run() method
+    mock_document = Document(id="0", content="test")
+    retriever = MockFilterRetriever(document_store=document_store_with_docs, mock_document=mock_document)
+    result = retriever.run(root_node="Query", query="", filters={})
+    assert result[0]["documents"][0] == mock_document
```

Comment:
Looks good, thank you! You can use the `document_store` fixture instead of `document_store_with_docs`, should be a bit faster.

Reply:
Done!